tests/test_presto_client.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import socket
+import os
+import json
+from mock import MagicMock, patch, PropertyMock
+from httplib import HTTPException, HTTPConnection
+from unittest import TestCase
+
+from package.scripts.presto_client import smoketest_presto, PrestoClient, \
+    InvalidArgumentError, URL_TIMEOUT_MS
+
+class TestPrestoClientSmoketest(TestCase):
+
+    nation = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14,
+              15, 16, 17, 18, 19, 20, 21, 22, 23, 24]
+
+    def setUp(self):
+        self.presto_client_mock = MagicMock()
+
+    def test_presto_queried_on_smoketest(self):
+        self.presto_client_mock.get_rows.return_value = TestPrestoClientSmoketest.nation
+
+        smoketest_presto(self.presto_client_mock)
+
+        assert self.presto_client_mock.execute_query.called
+
+    def test_failure_when_fewer_than_25_rows_returned_from_nation(self):
+        self.presto_client_mock.get_rows.return_value = [1]
+
+        TestCase.assertRaises(self, RuntimeError, smoketest_presto, self.presto_client_mock)
```

Comment:
assertRaises will only check that an exception of the given type was raised whereas assertRaisesRegexp will actually check the error message as well